cpp/oneapi/dal/algo/pca/backend/cpu/call_daal_pca_transform.hpp

```diff
@@ -0,0 +1,86 @@
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <daal/src/algorithms/pca/transform/pca_transform_kernel.h>
+
+#include "oneapi/dal/algo/pca/backend/cpu/infer_kernel.hpp"
+#include "oneapi/dal/backend/interop/common.hpp"
+#include "oneapi/dal/backend/interop/error_converter.hpp"
+#include "oneapi/dal/backend/interop/table_conversion.hpp"
+#include "oneapi/dal/algo/pca/infer_types.hpp"
+#include "oneapi/dal/table/row_accessor.hpp"
+
+namespace oneapi::dal::pca::backend {
+
+using std::int64_t;
+using dal::backend::context_cpu;
+
+namespace daal_pca_tr = daal::algorithms::pca::transform;
+namespace interop = dal::backend::interop;
+
+template <typename Float, daal::CpuType Cpu>
+using daal_pca_transform_kernel_t =
+    daal_pca_tr::internal::TransformKernel<Float, daal_pca_tr::Method::defaultDense, Cpu>;
+
+template <typename Float, typename Task>
+static infer_result<Task> call_daal_kernel(const context_cpu& ctx,
+                                           const descriptor_base<Task>& desc,
+                                           const table& data,
+                                           const model<Task>& model) {
+    const int64_t row_count = data.get_row_count();
+    const int64_t column_count = data.get_column_count();
+    const int64_t component_count = desc.get_component_count();
+
+    auto arr_data = row_accessor<const Float>{ data }.pull();
+    auto arr_eigvec = row_accessor<const Float>{ model.get_eigenvectors() }.pull();
+
+    auto arr_result = array<Float>::empty(row_count * component_count);
+
+    // TODO: read-only access performed with deep copy of data since daal numeric tables are mutable.
+    // Need to create special immutable homogen table on daal interop side
+
+    // TODO: data is table, not a homogen_table. Think better about accessor - is it enough to have just a row_accessor?
+    const auto daal_data =
+        interop::convert_to_daal_homogen_table(arr_data, row_count, column_count);
+    const auto daal_eigenvectors =
+        interop::convert_to_daal_homogen_table(arr_eigvec, component_count, column_count);
+    const auto daal_result =
+        interop::convert_to_daal_homogen_table(arr_result, row_count, component_count);
+
+    interop::status_to_exception(dal::backend::dispatch_by_cpu(ctx, [&](auto cpu) {
+        return daal_pca_transform_kernel_t<
+                   Float,
+                   oneapi::dal::backend::interop::to_daal_cpu_type<decltype(cpu)>::value>()
+            .compute(*daal_data.get(),
+                     *daal_eigenvectors.get(),
+                     NULL,
+                     NULL,
+                     NULL,
```

Comment:
Is it okay to use `NULL` instead of `nullptr`?

Reply:
like in `cpp/daal/src/algorithms/pca/transform/pca_transform_container.h`

Reply:
@rlnx Is it OK?

Reply:
fixed